features/admin.claims.v1/components/edit/local-claim/edit-basic-details-local-claims.tsx

```diff
@@ -957,18 +1056,124 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                         hint={ t("claims:local.forms.descriptionHint") }
                         readOnly={ isSubOrganization() || isReadOnly }
                     />
-                    <Field.Checkbox
-                        ariaLabel={ t("claims:local.forms.multiValued.label") }
-                        name="multiValued"
-                        label={ t("claims:local.forms.multiValued.label") }
-                        required={ false }
-                        defaultValue={ claim?.multiValued }
-                        data-testid={ `${testId}-form-multi-valued-input` }
-                        hint={ isSystemClaim
-                            ? t("claims:local.forms.multiValuedDisabledHint")
-                            : t("claims:local.forms.multiValuedHint") }
-                        readOnly={ isSubOrganization() || isSystemClaim || isReadOnly }
+                    <Field.Dropdown
+                        ariaLabel={ t("claims:local.forms.dataType.label") }
+                        name="dataType"
+                        label={ t("claims:local.forms.dataType.label") }
+                        data-testid={ `${testId}-form-data-type-input` }
+                        hint={ t("claims:local.forms.dataType.hint") }
+                        disabled={ isSubOrganization() || isSystemClaim || isReadOnly }
+                        options={ dataTypeOptions }
+                        value={ dataType }
+                        onChange={ (
+                            event: React.SyntheticEvent<HTMLElement, Event>,
+                            data: { value: string }
+                        ) => {
+                            setSubAttributes([]);
+                            setCanonicalValues([]);
+                            setDataType(data.value);
+                        } }
                     />
+                    { dataType === DataType.COMPLEX && (
+                        <>
+                            <Field.Dropdown
+                                ariaLabel="subAttributes-dropdown"
+                                name="subAttributesDropdown"
+                                label={ t("claims:local.forms.subAttributes.label") }
+                                placeholder={ t("claims:local.forms.subAttributes.placeholder") }
+                                options={
+                                    fetchedAttributes
```

Comment:
Let's do this in a useMemo to avoid computation on each render.

Reply:
Fixed by - https://github.com/wso2/identity-apps/pull/8111/commits/fc575885da12bc341fd55470310378ecbc3dc6ad